docs/src/piccolo/v1/index.rst

```diff
@@ -0,0 +1,42 @@
+.. _PiccoloV1:
+
+
+About Piccolo v1
+================
+
+**17th July 2023**
+
+Piccolo started in August 2018, and as of this writing is close to 5 years old.
+
+During that time we've had very few, if any, breaking changes. Stability has
+always been very important, as we rely on it for our production apps.
+
+So why release v1 now? We probably should have released v1 several years ago,
+but such are things. We now have some unavoidable breaking changes due to one
+of our main dependencies (Pydantic) releasing v2.
+
+In v2, the core of Pydantic has been rewritten in Rust, and has impressive
+improvements in performance. Likewise, other libraries in the ecosystem (such
+as FastAPI) have moved to Pydantic v2. It only makes sense that Piccolo does it
+too.
+
+In terms of your own code, you shouldn't see much difference. We removed the
+``pydantic_config_class`` from ``create_pydantic_model``, and replaced it with
+``pydantic_config``, but that's about it.
+
+However, quite a bit of internal code in Piccolo and its sister libraries
+`Piccolo API <https://piccolo-api.readthedocs.io>`_ and
+`Piccolo Admin <https://piccolo-admin.readthedocs.io>`_ had to be changed to
+support Pydantic v2. Supporting both Pydantic v1 and Pydantic v2 would be quite
+burdensome.
+
+So Piccolo v1 will just use Pydantic v2 and above.
+
+If you can't upgrade to Pydantic v2, then pin your Piccolo version to ``0.108.0``.
+You can find the `docs here for 0.108.0 <https://piccolo-orm.readthedocs.io/en/v0.108.0/>`_.
```

Comment:
I have one more question. Why isn't it pinned to `v0.118.0` as the last version with Pydantic v1 because there are some useful features between `v0.108.0` and `v0.118.0` that should work?

Reply:
Good catch - that's a typo. It should be `v0.118.0`!

Reply:
I've fixed this.

Reply:
Great. What about `**params` outside of `json_schema_extra`? When I did that, I got a lot of deprecated warnings when running the tests until I put `**params` inside `json_schema_extra`. Can you check that if it's not a problem?

Reply:
I'll have a look. It think what happened with Pydantic v1 is if you had this:

```python
Field(foo='bar')
```

It just added `foo` to the JSON schema. That's not allowed any more, which triggers the warnings.

But if the the argument is valid - for example `default`, then it should be OK.

https://github.com/pydantic/pydantic/blob/d249af9c7895020fee8e34f59ffb5c9220453719/pydantic/fields.py#L639C1-L640


For example:

```python
params = {'default': None'}
# This is OK
Field(**params)
# As it's equivalent to:
Field(default=None)
# And default is a permitted argument
```

I'll do some testing though to make sure.

Reply:
Thanks. I had that problem with unpacked params outside `json_schema_extra`. I'd check it out, but I don't have a laptop.

Reply:
You're right. I had issues with deprecated warnings because I left a `nullable` field in params kwarg, which is not allowed in Pydantic V2. You moved the `nullable` field to the `extra` kwarg located inside the `json_schema_extra` and everything is fine. I'm sorry I didn't notice it before, but check it out anyway.

Reply:
Ah, cool - thanks for checking. Will run the tests once more to check everything is ok, then wil merge in.